PhotonVision infrastructure -> poseestimator

GilStein1 opened the pull request:
⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡿⠿⠛⠻⠿⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠟⣋⣵⣾⣿⣿⣿⣿⣿⣶⣭⣛⢿⣿⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡟⣴⣿⣿⣿⡿⠿⠿⠟⠛⠋⠹⠿⣿⣦⢹⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡏⡴⠿⠛⠉⠐⠐⢉⣐⣒⣫⣥⣤⣽⣿⣿⡈⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⡿⠿⠿⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡇⣿⣶⣴⣶⣶⣿⣿⣿⣯⣭⣭⣴⣬⣽⣿⡇⣿⣿⣿⣿
⣿⣿⣿⠟⣫⣵⣶⣿⣿⣷⣶⣯⣝⠻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡇⢻⣟⣭⣷⡞⠭⣿⡿⠏⠘⠉⢙⣿⣿⣿⣇⣿⣿⣿⣿
⣿⠟⣱⣾⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣶⡝⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡸⣿⣇⣀⣀⣀⢹⡇⣷⣶⣶⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣡⣾⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣷⡙⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠉⠻⠿⢿⣿⣿⣿⣿⣇⢻⣿⣿⣿⡿⢸⣿⡘⢿⣿⣿⣿⣿⣿⡟⣿⣿⣿⣿
⣿⣤⠶⠖⠂⣂⣤⠭⢭⣭⣙⣛⡛⠛⠛⠛⡿⢿⣎⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣷⢰⣿⣷⣶⣬⣙⣛⣛⢸⢿⣿⣿⣧⣛⣛⣋⣼⣿⣿⣿⣿⠟⠀⠹⣿⣿⣿
⣿⢀⡤⠾⠷⣶⣶⣭⣲⣶⣖⣒⣂⣽⣝⣒⡈⣲⣿⡆⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣏⣚⣛⠿⣿⣿⣿⣿⣿⣧⡈⣿⡿⠋⠇⠀⠟⢈⠻⠻⡟⠁⠀⠀⣴⣮⣍⣛
⣿⣷⠾⣿⣷⣮⣻⣿⣿⣿⣿⣿⣿⣟⣩⣭⣍⢻⣿⣿⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣮⡛⢿⣿⣿⣿⣿⠀⠇⠘⠀⠀⠀⠀⢱⡆⠀⠀⢀⣤⣿⣿⣿⣿
⣠⡖⡀⠤⢤⣌⠩⣍⠩⣭⣭⢉⣩⠍⣀⣀⠩⣙⡛⢛⠘⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣦⡙⢿⣿⣿⣿⣄⠐⣦⣀⣀⣴⣿⡏⠁⣀⣾⣿⣿⣿⣿⣿
⣿⣿⢄⣀⣀⡘⢀⡿⠀⣩⡍⠘⠃⠀⠀⢐⣱⣿⣿⠸⠀⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣦⡻⣿⣿⣿⣦⣠⠉⠛⠉⠙⢀⣰⣿⣿⣿⣿⣿⣿⣿
⠙⠻⠶⠯⠽⠶⠛⣀⢀⣿⣿⡄⠄⠢⠭⠽⠷⠟⠋⣼⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣮⢻⣿⣿⣿⣿⣶⣾⣾⣾⣿⣿⣿⣿⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⣿⡟⣸⣿⣿⣿⢿⣿⣿⣿⣿⣿⣿⡇⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡈⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⣿⣇⠿⠿⡿⢟⣸⣿⣿⣿⣿⣿⣿⠇⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣧⠹⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣿⣿⣿⣿⣿⣿⣿⡟⡹⠷⡾⢿⣿⣿⣿⣿⣿⣿⣿⠀⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣇⢻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣿⣿⣿⣿⣿⡟⢡⡾⠃⣐⣋⠈⠛⢿⣿⣿⣿⡟⠁⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠘⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⠿⣿⣿⣿⡿⢁⠟⠀⠀⠀⠈⢱⠠⠈⣿⠿⠋⠉⢠⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡇⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⠃⠘⣿⣿⡇⢸⡀⡄⠀⠀⠀⢸⣶⡦⠻⠠⠀⠀⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣷⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⠀⢣⠡⢹⢁⡀⠓⣹⣦⣤⣴⡇⣿⠇⣤⠄⠀⣰⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⢄⠀⡆⠇⢸⣤⡜⢣⠩⣍⠩⣾⠈⡌⠉⡀⠃⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⢸⣦⣦⣐⠀⡟⠃⡀⢁⢋⠀⢿⠂⠀⣠⡼⠃⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⢸⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿
⣾⣿⣿⣿⢠⢀⡁⠄⠀⠐⠀⢈⣀⢲⣿⡇⣧⢻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡟⣼⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿

Sh-Pe committed:
Shahar: base starct

Sh-Pe committed:
Shahar: some changes

Sh-Pe committed:
Shahar: spotless

Sh-Pe committed:
Shahar: filters

Sh-Pe committed:
Shahar: the whole class in 3min

Sh-Pe committed:
Shahar: finished

GilStein1 committed:
Gil - changes

GilStein1 committed:
Gil - changes to the camera configurations

GilStein1 committed:
Gil - now calculating notes

GilStein1 committed:
Gil - filtered out the non apriltag observations

YoniKiriaty commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonObservationFiltered.java:
```diff
@@ -0,0 +1,76 @@
+		ArrayList<PhotonPoseData> output = new ArrayList<>();
+		for (GBPhotonCamera camera : cameras) {
+			Optional<PhotonPoseData> bestTarget = camera.getBestTargetData();
+			bestTarget.ifPresent(bestTargetData -> {
+				if (bestTargetData.target() == PhotonTarget.APRIL_TAG) {
+					output.add(bestTargetData);
+				}
+			});
```
i think it would be better to do if

GilStein1 replied:
no

hillelv commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonTarget.java:
```diff
@@ -0,0 +1,7 @@
+
+public enum PhotonTarget {
+	APRIL_TAG,
+	NOTE;
```
```suggestion
	GAME_OBJECT;
```

YoniKiriaty commented on src/main/java/frc/robot/poseestimator/photonvision/photoncamera/GBPhotonCamera.java:
```diff
@@ -0,0 +1,79 @@
+	private Optional<Pose3d> calculateNotePoseToRobot(PhotonTrackedTarget bestTarget) {
+		Transform3d noteToCamera = bestTarget.getBestCameraToTarget().inverse();
+		Transform3d noteToRobot = noteToCamera.plus(cameraToRobot);
+		Pose3d notePose = new Pose3d(noteToRobot.getX(), noteToRobot.getY(), noteToRobot.getZ(), noteToRobot.getRotation());
```
`.getTranslation()`

GilStein1 replied:
there is no constructor that gets a translation

GilStein1 commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonConstants.java:
```diff
@@ -0,0 +1,23 @@
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Transform3d;
+
+public class PhotonConstants {
```
a photon is both a particle and a wave so its kind of both and non

hillelv replied:
![image](https://github.com/user-attachments/assets/81b3ed9d-44e6-4a31-983a-2ce89f258831)

GilStein1 committed:
Gil - renaming

GilStein1 committed:
Gil - somthing and spotless

hillelv commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonObservationFiltered.java:
```diff
@@ -0,0 +1,73 @@
+	}
+
+	private VisionObservation calculateObservationFromPoseData(PhotonPoseData poseData) {
+		Pose3d robot3DPose = poseData.robotPose();
+		Pose2d robot2DPose = new Pose2d(
```
```suggestion
		Pose3d robotPose3d = poseData.robotPose();
		Pose2d robotPose2d = new Pose2d(
```

hillelv commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonObservationFiltered.java:
```diff
@@ -0,0 +1,73 @@
+		Pose2d robot2DPose = new Pose2d(
+			robot3DPose.getX(),
+			robot3DPose.getY(),
+			Rotation2d.fromRadians(robot3DPose.getRotation().getZ())
```
```suggestion
			new Rotation2d(robot3DPose.getRotation().getZ())
```

hillelv commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonVisionCamera.java:
```diff
@@ -0,0 +1,75 @@
+	private Optional<Pose3d> calculateGameObjectPoseToRobot(PhotonTrackedTarget bestTarget) {
+		Transform3d gameObjectToCamera = bestTarget.getBestCameraToTarget().inverse();
+		Transform3d gameObjectToRobot = gameObjectToCamera.plus(cameraToRobot);
+		Pose3d notePose = new Pose3d(gameObjectToRobot.getTranslation(), gameObjectToRobot.getRotation());
```
```suggestion
		Pose3d gameObjectPose = new Pose3d(gameObjectToRobot.getTranslation(), gameObjectToRobot.getRotation());
```

YoniKiriaty commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonObservationFiltered.java:
```diff
@@ -0,0 +1,73 @@
+	private final ArrayList<PhotonVisionCamera> cameras;
+
+	public PhotonObservationFiltered(CameraConfiguration[] cameraConfigurations) {
+		cameras = new ArrayList<>();
```
this.

YoniKiriaty commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonObservationFiltered.java:
```diff
@@ -0,0 +1,73 @@
+	public PhotonObservationFiltered(CameraConfiguration[] cameraConfigurations) {
+		cameras = new ArrayList<>();
+		for (CameraConfiguration cameraConfiguration : cameraConfigurations) {
+			cameras.add(new PhotonVisionCamera(cameraConfiguration));
```
this.cameras

YoniKiriaty commented on src/main/java/frc/robot/poseestimator/photonvision/PhotonVisionCamera.java:
```diff
@@ -0,0 +1,75 @@
+	private final PhotonVisionTarget target;
+
+	public PhotonVisionCamera(String logPath, String cameraName, Transform3d cameraToRobot, PhotonVisionTarget target) {
+		super(logPath + cameraName + "/");
```
maybe also add super.setName("photonVisionCamera" + name)

GilStein1 committed:
Gil - cr stuff

GilStein1 committed:
Gil & shahar - idk gradle what?

Sh-Pe commented:
can't simulate

GilStein1 committed:
Gil - gradle is not gradle

itamaroryan committed:
Shahar: logging

itamaroryan committed:
Shahar: changes

Sh-Pe committed:
Shahar: generalizatoin

Sh-Pe committed:
Shahar: small improvments

Sh-Pe committed:
Shahar: improvements

Sh-Pe committed:
Shahar: refactoring

Sh-Pe committed:
Shahar: spotless

GilStein1 committed:
Gil - splitted the filters class to 3 classes, each representing a type of target filter

YoniKiriaty commented on src/main/java/frc/robot/poseestimator/photonvision/photonvisionfilters/PhotonVisionFiltered.java:
```diff
@@ -0,0 +1,47 @@
+	public double[] getStandardDeviations(PhotonTargetRawData targetData) {
+		double ambiguity = targetData.ambiguity();
+		return new double[] {
+			ambiguity / PhotonVisionConstants.AMBIGUITY_TO_LOCATION_STANDARD_DEVIATIONS_FACTOR,
+			ambiguity / PhotonVisionConstants.AMBIGUITY_TO_LOCATION_STANDARD_DEVIATIONS_FACTOR,
+			ambiguity / PhotonVisionConstants.AMBIGUITY_TO_ROTATION_STANDARD_DEVIATIONS_FACTOR};
```
save variable

GilStein1 committed:
Gil - ooga booga

GilStein1 committed:
Gil - removed unnesecery PhotonVisionTarget from raw data

GilStein1 committed:
Gil - renaming

GilStein1 committed:
Gil - logging

Sh-Pe committed:
Shahar: now max

Sh-Pe committed:
Shahar: logging moved to base PhotonVisionFiltered class and name method added to the record

Sh-Pe committed:
Shahar: same but it uses the filtered data

Sh-Pe committed:
Shahar: graldew

Sh-Pe committed:
Shahar: cameas log path prefix is now constant

Sh-Pe committed:
Shahar: chagnes apply to the other class

GilStein1 committed:
Gil - removed unused arguments

GilStein1 committed:
Gil - big P

Yotam7 committed:
change line endings on gradew.bat to be same as poseestimator

greenblitz4590 committed:
Gil & Shahar - temp code, will be fixed!

Sh-Pe committed:
Shahar: improvements

Sh-Pe committed:
Shahar: spotless

Sh-Pe committed:
Shahar: tranform, spotless, tolerences

Sh-Pe committed:
Shahar: added support for multitargets within one pipeline

Sh-Pe committed:
Shahar: spotless

Sh-Pe committed:
conflict solving before the merging

Merge remote-tracking branch 'origin/poseestimator' into photon-vision-april-tags

Sh-Pe committed:
Shahar: merging & refactoring to the right directory

hillelv commented on src/main/java/frc/robot/RobotManager.java:
```diff
@@ -24,6 +24,7 @@ public class RobotManager extends LoggedRobot {
 
 	private Robot robot;
 
+
```
remove

Sh-Pe committed:
Merge remote-tracking branch 'origin/poseestimator' into photon-vision-april-tags

Sh-Pe committed:
Shahar: wip

Sh-Pe committed:
Shahar: generalizations

Sh-Pe committed:
Shahar: generalization almost finished

Sh-Pe committed:
Shahar: turns out is never ends

Sh-Pe committed:
Shahar: I think it's finished now

Sh-Pe committed:
Shahar: refactoring for more consistent code

Sh-Pe committed:
Shahar: better way to get limelight's latency

Sh-Pe committed:
Shahar: better logging, refactoring

GilStein1 commented on src/main/java/frc/robot/vision/aprilTags/AprilTagFilters.java:
```diff
@@ -0,0 +1,65 @@
+import edu.wpi.first.math.geometry.*;
+import frc.robot.vision.VisionRawData;
+
+public class AprilTagFilters {
```
```suggestion
public class VisionPoseEstimationFilters {
```

GilStein1 commented on src/main/java/frc/robot/vision/aprilTags/AprilTagFiltersTolerances.java:
```diff
@@ -1,12 +1,14 @@
 import edu.wpi.first.math.geometry.Rotation2d;
 
-public record LimelightFiltersTolerances(
+public record AprilTagFiltersTolerances(
```
```suggestion
public record VisionPoseEstimationFiltersTolerances(
```

GilStein1 commented on src/main/java/frc/robot/vision/aprilTags/AprilTagFilters.java:
```diff
@@ -0,0 +1,65 @@
+
+public class AprilTagFilters {
+
+	public static boolean keepRobotPoseData(VisionRawData visionData, Pose2d currentEstimatedPose, AprilTagFiltersTolerances tolerances) {
+		return AprilTagFilters.isLimelightOutputInTolerance(
+			visionData,
+			currentEstimatedPose,
+			tolerances.normalizedPositionTolerance(),
+			tolerances.normalizedRotationTolerance()
+		)
+			&& AprilTagFilters.isRollInTolerance(visionData, tolerances.rollTolerance())
+			&& AprilTagFilters.isPitchInTolerance(visionData, tolerances.pitchTolerance())
+			&& AprilTagFilters.isRobotOnGround(visionData, tolerances.robotToGroundToleranceMeters())
+			&& !AprilTagFilters.isDataTooAmbiguous(visionData, tolerances.maximumAmbiguity())
+			&& !AprilTagFilters.isLatencyTooHigh(visionData, tolerances.maximumLatency());
+	}
```
again, I think there should be separate classes for the filters of the photon and limelight. There could be filters that would only fit photon and will not fit limelight. Also, we can't get the ambiguity of the limelights

GilStein1 commented on src/main/java/frc/robot/vision/limelights/LimelightEntryValue.java:
```diff
@@ -8,7 +8,7 @@ public enum LimelightEntryValue {
 	PITCH_ANGLE(4),
 	YAW_ANGLE(5),
-	TOTAL_LATENCY(6);
+	TOTAL_LATENCY_ROBOT_POSE_ONLY(6);
```
? 
TOTAL_ROBOT_POSE_LATENCY?

Sh-Pe committed:
Shahar: formatting

GilStein1 commented on src/main/java/frc/robot/vision/limelights/MultiLimelights.java:
```diff
@@ -26,27 +29,35 @@ public void updateGyroAngles(GyroAngleValues gyroAngleValues) {
 	}
 
+	// we should find a better function later
+	private double calculateAmbiguity(double aprilTagHeightMeters) {
+		double heightPower3 = Math.pow(aprilTagHeightMeters, 3);
+		return heightPower3 / (heightPower3 + 200);
+	}
```
we can't calculate ambiguity from distance to apriltag, that's not the same and I don't think they correlate

GilStein1 reviewed (commented):
I really think that there should be an ILimelightFilterer and an IPhotonFilterer. We can't just treat them as the same thing, they each have their own filters

GilStein1 commented on src/main/java/frc/robot/vision/photonvision/photonvisionfilters/PhotonVisionAprilTagFilters.java:
```diff
@@ -0,0 +1,49 @@
+		return output;
+	}
+
+	private Pose2d calculateCurrentPoseEstimation() {
+		return getEstimatedPoseAtTimestamp.apply(Conversions.microSecondsToSeconds(Logger.getRealTimestamp()));
+	}
```
you need the pose estimation at the time of the observation, not the current time. get that timestamp from the photon camera

GilStein1 commented on src/main/java/frc/robot/vision/photonvision/photonvisionfilters/PhotonVisionAprilTagFilters.java:
```diff
@@ -0,0 +1,49 @@
+
+	@Override
+	protected boolean keepPhotonVisionData(VisionRawData visionData) {
+		return AprilTagFilters.keepRobotPoseData(visionData, calculateCurrentPoseEstimation(), tolerances);
```
the photon observations should not be filtered using the filters used on the limelight observations

Sh-Pe replied:
these are the same filters

GilStein1 reviewed (commented):
good job with the abstract PhotonVisionFiltered class

Sh-Pe commented on src/main/java/frc/robot/vision/VisionRawData.java:
```diff
@@ -0,0 +1,5 @@
+
+import edu.wpi.first.math.geometry.Pose3d;
+
+public record VisionRawData(String cameraName, Pose3d targetPose, double ambiguity, double timestamp, double latency) {}
```
in this case, the robot is the target. But, this could also be an Note.

Sh-Pe commented on src/main/java/frc/robot/vision/VisionRawData.java:
```diff
@@ -0,0 +1,5 @@
+
+import edu.wpi.first.math.geometry.Pose3d;
+
+public record VisionRawData(String cameraName, Pose3d targetPose, double ambiguity, double timestamp, double latency) {}
```
it does has latency, and I think we can calculate the ambiguity

Sh-Pe committed:
Shahar: CR

Sh-Pe commented:
A new pull request, interfacing with the latest abstractions in the `poseestimator-rewrite` branch, will soon be released.
